src/persist-client/src/internal/compact.rs

```diff
@@ -740,7 +740,17 @@ where
             }
             tokio::task::yield_now().await;
         }
-        let batch = batch.finish(&real_schemas, desc.upper().clone()).await?;
+        let mut batch = batch.finish(&real_schemas, desc.upper().clone()).await?;
+        // Use compaction as a method of getting inline writes out of state, to
+        // make room for more inline writes.
```

Comment:
Setting the `BatchBuilderConfig::inline_update_threshold_bytes` to 0 during compaction would have the same effect. But I guess we need `flush_to_blob` for the CaA path in any case...

Reply:
Oh I like that a lot. It lets BatchBuilder do its normal pipelining of writes. Done (and then changed this to a defensive check)